zjit/src/hir.rs

```diff
@@ -2271,8 +2277,21 @@ impl Function {
             for insn_id in old_insns {
                 if let send @ Insn::SendWithoutBlock { recv, .. } = self.find(insn_id) {
                     let recv_type = self.type_of(recv);
-                    if reduce_to_ccall(self, block, recv_type, send, insn_id).is_ok() {
-                        continue;
+                    match reduce_to_ccall(self, block, recv_type, send, insn_id) {
+                        Ok(()) => continue,
+                        Err(Some(cme)) => {
+                            let owner = unsafe { (*cme).owner };
+                            let called_id = unsafe { (*cme).called_id };
+                            let class_name = get_class_name(owner);
+                            let method_name = called_id.contents_lossy();
+                            let qualified_method_name = format!("{}#{}", class_name, method_name);
+                            let unoptimized_cfunc_counter_pointers = ZJITState::get_unoptimized_cfunc_counter_pointers();
+                            let counter_ptr = unoptimized_cfunc_counter_pointers.entry(qualified_method_name.clone()).or_insert_with(|| Box::new(0));
+                            let counter_ptr = &mut **counter_ptr as *mut u64;
+
+                            self.push_insn(block, Insn::IncrDynamicCounter { counter_ptr });
```

Comment:
Can we insert this only if `get_option!(stats)` is true? It seems inconsistent with `Insn::IncrCounter`. It also seems inconsistent to insert them in HIR when it does nothing.

Reply:
Ah I got the wrong impression that we're always generating `IncrCounter`. Updated